app/questionnaire/questionnaire_schema.py

```diff
@@ -734,6 +726,18 @@ def is_answer_in_repeating_section(self, answer_id: str) -> bool | None:
     def is_answer_dynamic(self, answer_id: str) -> bool:
         return answer_id in self._dynamic_answer_ids
 
+    def get_list_name_for_dynamic_answer(self, block_id: str) -> str | None:
+        return self.get_block(block_id).get("question", {}).get("dynamic_answers", {}).get("values", {}).get("identifier", {})  # type: ignore
+        # type ignore block always exists at this point
+
+    def block_has_dynamic_answer(self, block_id: str) -> bool:
+        return (
+            "dynamic_answers" in self.get_block(block_id).get("question", {})  # type: ignore
```

Comment:
If this is being type ignored as the block will always exist, this might as well be simplified to` return "dynamic_answers" in self.get_block(block_id).get("question", {})  # type: ignore`